development/language/usage.rst

```diff
@@ -8,48 +8,96 @@ also a requirement for Extensions to use said system for all output presented to
+    // Output: "My translation has 1 parameter"
+
+.. warning::
+
+    When using ``%d`` for plurals make sure to allow translators to use plurals
+    according to their language. For more information see: :doc:`plurals`
+
+.. note::
 
-To lookup a language entry inside of phpBB, query the ``$user`` object like
-this: ``$user->lang('LANG_KEY')``, which will return the language string
-assigned to the key ``LANG_KEY`` in the current user's language. Note that many
-functions in phpBB automatically translate their arguments; passing the language
-key as an argument is usually enough.
+    Many functions in phpBB automatically translate their arguments; passing the
+    language key as an argument is usually enough.
 
 The dictionary is not loaded into memory all at once, but is split up in several
 files, which have to be manually loaded on demand. The files are stored inside
 the language directory, named after its
-`language tag <https://area51.phpbb.com/docs/30x/coding-guidelines.html#translation>`_,
+`language tag <https://area51.phpbb.com/docs/master/coding-guidelines.html#translation>`_,
 where one subdirectory is used for each installed language. The default language
 files are in the root of that subdirectory, while ACP language files go into the
 ``/acp`` subdirectory; email templates are placed in the ``/email``
-subdirectory. Files for extensions should be placed in the extension's
-``/language`` directory.
+subdirectory.
+
+.. note::
+    Files for extensions should be placed in the extension's ``/language``
+    directory.
+
+Loading language files
+======================
+
+Loading from phpBB core
+-----------------------
 
 Additional language files can be loaded during user setup via the call to the
-``$user->setup()`` method at the top of each accessible file. To do so, pass the
-name with the subdirectory but without extension as argument of setup. For
-instance ``$user->setup('search')`` or ``$user->setup('mylangfile')`` or
-``$user->setup(array('mylangfile', 'search'))``. If you need to add a language
-file at a later stage, use add_lang, e.g.
-``$user->add_lang(array('mylangfile', 'search'))``.
+``phpbb\user::setup()`` method at the top of each accessible file. To do so,
+pass the name with the subdirectory but without extension as argument of setup.
+
+.. code-block:: php
 
-A sightly more difficult situation arises when the displayed text should contain
-links etc. phpBB uses a ``lang()``-method for that, for instance:
+    $user->setup('search')
+    // or
+    $user->setup('ucp')
+    // or
+    $user->setup(array('ucp', 'search'))
+
+Since ``phpbb\user::setup()`` must only be called once,
+``phpbb\language\language::add_lang()`` has to be used, to load additional
+language files, after ``phpbb\user::setup()`` has already been called.
+
+Loading from an extension
+-------------------------
+
+To load a file from an extension
+you need to use ``phpbb\language\language::add_lang_ext()`` which takes
```

Comment:
``phpbb\language\language::add_lang()`` first argument array of the language files, second argument is the name of the extension